### PR TITLE
[SPARK-36190][PYTHON] Improve the rest of DataTypeOps tests by avoiding joins

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Improve the rest of DataTypeOps tests by avoiding joins.

### Why are the changes needed?
bool, string, numeric DataTypeOps tests have been improved by avoiding joins.
We should improve the rest of the DataTypeOps tests in the same way.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Unit tests.

Keyword:SPARK-35337

### CODE REVIEW

**Comment by xinrong-databricks** [comment]
CC @ueshin @HyukjinKwon @itholic @Yikun

**Review by HyukjinKwon — APPROVED** [review]

**Review by itholic — APPROVED** [review]

File: python/pyspark/pandas/tests/data_type_ops/test_complex_ops.py [hunk 1]
```diff
@@ -252,70 +262,76 @@ def test_invert(self):
...
-        with option_context("compute.ops_on_diff_frames", True):
-            self.assert_eq(
-                self.pser == self.other_pser, (self.psser == self.other_psser).sort_index()
-            )
-            self.assert_eq(self.pser == self.pser, (self.psser == self.psser).sort_index())
-            self.assert_eq(
-                self.struct_pser == self.struct_pser,
-                (self.struct_psser == self.struct_psser).sort_index(),
-            )
+        pdf, psdf = self.complex_pdf, self.complex_pdf
```
Comment (Yikun) [inline python/pyspark/pandas/tests/data_type_ops/test_complex_ops.py:265]: Not related this change, but looks like we could also add `this_struct == this_struct` case just like array. Definately, we can add it in separate followup.

**Review by Yikun — APPROVED** [review]
LGTM

File: python/pyspark/pandas/tests/data_type_ops/test_complex_ops.py (hunk 1 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/data_type_ops/test_complex_ops.py:265]: Test cases on that are added. Thanks!

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

**Comment by ueshin** [comment]
Thanks! merging to master/3.2

**Pull request closed without merging**